src/par_iter/test.rs

```diff
@@ -179,6 +179,57 @@ pub fn check_increment() {
 }
 
 #[test]
+pub fn check_skip() {
+    let a: Vec<usize> = (0..10).collect();
+
+    let mut b = vec![];
+    a.par_iter()
```

Comment:
Nit: I would do `let b: Vec<_> = a.par_iter()...collect();` and  then compare against the same iterator chain, but sequential (i.e., just check that par/seq get same result).

Also, we usually add a `weight_max()` which forces maximum splitting. It tends to expose bugs.